File: lab2/singleton/singleton.java
class Kot
{
    private static Kot jedyny = new Kot();
    private int nr = 0;
    private Kot()
    {
        nr++;
    }
    public static Kot DajEgzemplarz()
    {
        return jedyny;
        
    }
    public void daj_glos()
    {
        System.out.println("To ja singleton nr= "+nr);
    }
}

public class  singleton{
    public static void main(String[] Args)
    {
        
        Kot[]tablica = new Kot[10];

        for(int i = 0; i < tablica.length; i++)
        {
            tablica[i] = Kot.DajEgzemplarz();
        }
        
        for(Kot K:tablica)
            K.daj_glos();
    }
}

File: lab2/SingletonCpp/singleton.cpp
#include "singleton.h"

Kot *Kot::jedyny = new Kot();

Kot::Kot()
{
	nr++;
}

Kot *Kot::DajEgzemplarz()
{
	return jedyny;

}

void Kot::daj_glos()
{
	std::wcout << L"To ja singleton nr= " << nr << std::endl;
}

void singleton::main(std::vector<std::wstring> &Args)
{

	std::vector<Kot*> tablica(10);

	for (int i = 0; i < tablica.size(); i++)
	{
		tablica[i] = Kot::DajEgzemplarz();
	}

	for (auto K : tablica)
	{
		K->daj_glos();
	}
}
